src/main/java/org/folio/rest/support/ShelvingOrderUpdate.java

```diff
@@ -0,0 +1,197 @@
+package org.folio.rest.support;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.pgclient.PgConnection;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.Tuple;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dbschema.Versioned;
+import org.folio.rest.jaxrs.model.Item;
+import org.folio.rest.jaxrs.model.TenantAttributes;
+import org.folio.rest.persist.PgUtil;
+import org.folio.rest.persist.PostgresClient;
+
+public class ShelvingOrderUpdate {
+
+  private static final Logger log = LogManager.getLogger();
+
+  private static final String SQL_EXISTS_ITEMS_TO_UPDATE = new StringBuilder()
+      .append("SELECT EXISTS (SELECT null FROM item WHERE not (jsonb ? 'effectiveShelvingOrder'))")
+      .toString();
+
+  private static final String SQL_SELECT_ITEMS = new StringBuilder()
+      .append("SELECT id AS \"id\", ")
+      .append("jsonb AS \"itemJson\" ")
```

Comment:
Do we need entire item json, or just few properties from it?

Reply:
Hum, that's a good question from point of view reducing of acquired amount of data as maximum as possible, but as it was implemented in calculation of shelving order there is whole Item object required as parameter. This is quite reasonable because where is not only string values involved in calculation like volume, enumeration etc. but also composite data object like "effective call number components", so anyway we need to restore the whole Item from database and pass it for calculation or just retrieve separate values as string with part of Items JSON data, i.e. "effective call number component", for further deserialization and passing into already implemented shelving calculation routine. By the way, this routine should be also extended with overloaded methods with another signature too. So the main question here is - will be current retrieval of items JSON with further deserialization be less effective that retrieval of separate part of items which in turns needs to include part of items as composite JSON object too.